lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -546,6 +550,32 @@ class QcFlagService {
         };
     }
 
+    /**
+     * Find QC flags in GAQ effective periods for given data pass and run
+     *
+     * @param {number} dataPassId id od data pass
+     * @param {number} runNumber run number
+     * @return {Promise<GaqQcFlags>} promise of aggregated QC flags
+     */
+    async getGaqQcFlags(dataPassId, runNumber) {
+        const effectiveGaqPeriods = await QcFlagRepository.findFlagsInGaqEffectivePeriods({ dataPassId, runNumbers: [runNumber] });
+        const qcFlags = (await QcFlagRepository.findAll({
+            where: { id: { [Op.in]: effectiveGaqPeriods.flatMap(({ contributingFlagIds }) => contributingFlagIds) } },
+        })).map(qcFlagAdapter.toEntity);
+
+        const idToFlag = Object.fromEntries(qcFlags.map((flag) => [flag.id, flag]));
```

Comment:
why doing this?
Why not extracting a list of flag ids, then simply fetching it and using this as `contributingFlags` ?

Reply:
because, simply, one request can be done instead of many